SyntheticGeometry/ProjectiveLine.lagda.md

```diff
@@ -0,0 +1,252 @@
+Facts about the projective line ℙ¹
+==================================
+
+```agda
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Powerset using (_∈_; _⊆_) renaming (ℙ to Powerset)
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Functions.Embedding
+open import Cubical.Functions.Surjection
+open import Cubical.Functions.Involution
+
+open import Cubical.HITs.SetQuotients as SQ
+open import Cubical.HITs.PropositionalTruncation as PT using (∥_∥₁)
+open import Cubical.HITs.Pushout as Pushout
+
+open import Cubical.Data.FinData
+open import Cubical.Data.Sigma
+
+open import Cubical.Algebra.CommRing
+open import Cubical.Algebra.CommRing.LocalRing
+
+import SyntheticGeometry.SQC
+
+module SyntheticGeometry.ProjectiveLine
+  {ℓ : Level}
+  (k : CommRing ℓ)
+  (k-local : isLocal k)
+  (k-sqc : SyntheticGeometry.SQC.sqc-over-itself k)
+  where
+
+open import SyntheticGeometry.Spec k
+open import SyntheticGeometry.Open k
+open import SyntheticGeometry.Affine k k-local k-sqc
+open import SyntheticGeometry.ProjectiveSpace k k-local k-sqc
+open import SyntheticGeometry.SQC.Consequences k k-local k-sqc
+```
+
+Exhibit ℙ¹ as a pushout of two copies of 𝔸¹.
+
+```agda
+𝔸¹ˣ : Type ℓ
+𝔸¹ˣ = Σ[ x ∈ ⟨ k ⟩ ] x ∈ k ˣ
+
+module PushoutMaps
+  where
+  f g : 𝔸¹ˣ → ⟨ k ⟩
+  f (x , _) = x
+  g (_ , (x⁻¹ , _)) = x⁻¹
+
+open PushoutMaps
+
+ℙ¹-as-pushout : Type ℓ
+ℙ¹-as-pushout =
+  Pushout {A = 𝔸¹ˣ} {B = ⟨ k ⟩} {C = ⟨ k ⟩} f g
+
+module Comparison
+  where
+
+  open CommRingStr (snd k)
+  open Consequences k k-local
+  open Units k
+
+  𝔸² : Type ℓ
+  𝔸² = 𝔸ⁿ⁺¹ 1
+
+  𝔸²-0 : Type ℓ
+  𝔸²-0 = 𝔸ⁿ⁺¹-0 1
+
+  𝔸²-≡ :
+    {xy x'y' : 𝔸²} →
+    (xy zero ≡ x'y' zero) →
+    (xy one ≡ x'y' one) →
+    xy ≡ x'y'
+  𝔸²-≡ x≡x' y≡y' = funExt (λ{ zero → x≡x' ; one → y≡y'})
+
+  -- More specific types for some operations.
+  [_]ℙ¹ : 𝔸²-0 → ℙ 1
+  [_]ℙ¹ = [_]
+
+  inl' inr' : ⟨ k ⟩ → ℙ¹-as-pushout
+  inl' = inl
+  inr' = inr
+
+  -- The autoequivalence of 𝔸¹ˣ that turns f into g and vice versa.
+  inversion : 𝔸¹ˣ → 𝔸¹ˣ
+  inversion (x , x-inv) = (x ⁻¹) , RˣInvClosed x
+    where
+    instance
+      _ : x ∈ k ˣ
+      _ = x-inv
+
+  -- Just checking that this is definitional.
+  g≡f∘inversion : (x : 𝔸¹ˣ) → g x ≡ f (inversion x)
+  g≡f∘inversion x = refl
+
+  isEquiv-inversion : isEquiv inversion
+  isEquiv-inversion = involIsEquiv (λ x → Σ≡Prop (snd ∘ (k ˣ)) refl)
+
+  isSet-ℙ¹-as-pushout : isSet ℙ¹-as-pushout
+  isSet-ℙ¹-as-pushout =
+    Pushout.preserveHLevelEmbedding
+      f
+      g
+      {n = 0}
+      isEmbedding-f
+      (isEmbedding-∘ isEmbedding-f (isEquiv→isEmbedding isEquiv-inversion))
+      is-set
+      is-set
+    where
+    isEmbedding-f = snd (snd (Subset→Embedding (k ˣ)))
+
+  module Function
```

Comment:
Maybe a comment here and at the end of this module is better, since it is not really used as a module.

Reply:
Done.